src/app/_services/auth.service.ts

```diff
@@ -80,6 +80,12 @@ export class AuthService implements CanActivate {
     private injector: Injector,
     private angulartics: Angulartics2
   ) {
+
+    // For session timeout on user inactivity
+    this.idle.setIdle(IdleWatcherUtil.generateIdleTime()); // Set an idle time of 1 min, before starting to watch for timeout
+    this.idle.setTimeout(IdleWatcherUtil.generateSessionLength()); // Log user out after 90 mins of inactivity
+    this.idle.setInterrupts([new EventTargetInterruptSource(window, 'mousemove keydown DOMMouseScroll mousewheel mousedown touchstart touchmove scroll')]);
+
```

Comment:
Any reason for moving this block outside of `initIdleWatcher`?

Reply:
was just following the example mentioned in ng-idle GitHub page. This can be moved to where it is.

Reply:
Gotcha. It might help clean things up to gather all the idle "initialization" code in one function. Up to you.